test/jdk/javax/swing/JTable/PrintAllPagesTest.java

```diff
@@ -0,0 +1,156 @@
+                } catch (PrinterException e) {
+                    throw new RuntimeException("Printing failed: " + e);
+                }
+            });
+
+            // wait for latch to complete
+            if (!latch.await(5, TimeUnit.MINUTES)) {
+                throw new RuntimeException(" User has not executed the test");
+            }
+
+            if (!testResult) {
+                throw new RuntimeException("Only 1st page is printed out of multiple pages");
+            }
+        } finally {
+            SwingUtilities.invokeAndWait(() -> {
+                dispose();
+            });
+        }
+    }
+
+    private static void dispose() {
+        if (dialog != null) {
+            dialog.dispose();
+        }
+        if (f != null) {
+            f.dispose();
+        }
+    }
+
+    private static void printAllPagesTest() {
+        TableModel dataModel = new AbstractTableModel() {
+            @Override
+            public int getColumnCount() {
+                return 10;
+            }
+
+            @Override
+            public int getRowCount() {
+                return 1000;
+            }
+
+            @Override
+            public Object getValueAt(int row, int col) {
+                return Integer.valueOf(0 == col ? row + 1 : row * col);
+            }
+        };
+        table = new JTable(dataModel);
+        JScrollPane scrollpane = new JScrollPane(table);
+        table.scrollRectToVisible(table.getCellRect(table.getRowCount() - 1, 0, false));
+
+        f = new JFrame("Table test");
+        f.add(scrollpane);
+        f.setSize(1000, 800);
+        f.setUndecorated(true);
+        f.setLocationRelativeTo(null);
+        f.setDefaultCloseOperation(WindowConstants.EXIT_ON_CLOSE);
+        f.setVisible(true);
+    }
+
+    private static void createUI() {
+        String description
+                = " INSTRUCTIONS:\n"
+                + " A JTable with 1000 rows and a print dialog will be shown.\n"
+                + " If only 1 page is printed,\n "
+                + " then press fail else press pass";
+
+        dialog = new JDialog();
+        dialog.setTitle("textselectionTest");
+        JTextArea textArea = new JTextArea(description);
+        textArea.setEditable(false);
+        final JButton passButton = new JButton("PASS");
+        passButton.addActionListener((e) -> {
+            testResult = true;
+            latch.countDown();
+        });
+        final JButton failButton = new JButton("FAIL");
+        failButton.addActionListener((e) -> {
+            testResult = false;
+            latch.countDown();
+        });
+        JPanel mainPanel = new JPanel(new BorderLayout());
+        mainPanel.add(textArea, BorderLayout.CENTER);
+        JPanel buttonPanel = new JPanel(new FlowLayout());
+        buttonPanel.add(passButton);
+        buttonPanel.add(failButton);
+        mainPanel.add(buttonPanel, BorderLayout.SOUTH);
+        dialog.add(mainPanel);
+        dialog.setUndecorated(true);
```

Comment:
It doesn't prevent you from using Alt+F4 to close the window but makes experience worse.

Is it possible to use JFrame instead of JDialog to display instructions?

If I accidentally switch to another window which overlaps the dialog, there's no way to bring it to the top other than minimizing all the windows that might overlap the dialog. If it's a frame, it's displayed in the Taskbar.

Reply:
I will rather not make the test more complicated...it's not a test fix..it just to verify the product fix..
I am not able to close in windows using ALT+F4 anyway...
Other suggestions accepted...